chips/rp2040/src/timer.rs

```diff
@@ -0,0 +1,266 @@
+        (0x030 => pause: ReadWrite<u32>),
+        /// Raw Interrupts
+        (0x034 => intr: ReadWrite<u32, INTR::Register>),
+        /// Interrupt Enable
+        (0x038 => inte: ReadWrite<u32, INTE::Register>),
+        /// Interrupt Force
+        (0x03C => intf: ReadWrite<u32, INTF::Register>),
+        /// Interrupt status after masking & forcing
+        (0x040 => ints: ReadWrite<u32, INTS::Register>),
+        (0x044 => @END),
+    }
+}
+register_bitfields![u32,
+TIMEHW [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+TIMELW [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+TIMEHR [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+TIMELR [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+ALARM0 [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+ALARM1 [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+ALARM2 [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+ALARM3 [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+ARMED [
+    ARMED OFFSET(0) NUMBITS(4) []
+],
+TIMERAWH [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+TIMERAWL [
+    VALUE OFFSET (0) NUMBITS (32) []
+],
+DBGPAUSE [
+    /// Pause when processor 1 is in debug mode
+    DBG1 OFFSET(2) NUMBITS(1) [],
+    /// Pause when processor 0 is in debug mode
+    DBG0 OFFSET(1) NUMBITS(1) []
+],
+PAUSE [
+
+    PAUSE OFFSET(0) NUMBITS(1) []
+],
+INTR [
+
+    ALARM_3 OFFSET(3) NUMBITS(1) [],
+
+    ALARM_2 OFFSET(2) NUMBITS(1) [],
+
+    ALARM_1 OFFSET(1) NUMBITS(1) [],
+
+    ALARM_0 OFFSET(0) NUMBITS(1) []
+],
+INTE [
+
+    ALARM_3 OFFSET(3) NUMBITS(1) [],
+
+    ALARM_2 OFFSET(2) NUMBITS(1) [],
+
+    ALARM_1 OFFSET(1) NUMBITS(1) [],
+
+    ALARM_0 OFFSET(0) NUMBITS(1) []
+],
+INTF [
+
+    ALARM_3 OFFSET(3) NUMBITS(1) [],
+
+    ALARM_2 OFFSET(2) NUMBITS(1) [],
+
+    ALARM_1 OFFSET(1) NUMBITS(1) [],
+
+    ALARM_0 OFFSET(0) NUMBITS(1) []
+],
+INTS [
+
+    ALARM_3 OFFSET(3) NUMBITS(1) [],
+
+    ALARM_2 OFFSET(2) NUMBITS(1) [],
+
+    ALARM_1 OFFSET(1) NUMBITS(1) [],
+
+    ALARM_0 OFFSET(0) NUMBITS(1) []
+]
+];
+const TIMER_BASE: StaticRef<TimerRegisters> =
+    unsafe { StaticRef::new(0x40054000 as *const TimerRegisters) };
+
+pub struct RPTimer<'a> {
+    registers: StaticRef<TimerRegisters>,
+    client: OptionalCell<&'a dyn hil::time::AlarmClient>,
+}
+
+impl<'a> RPTimer<'a> {
+    pub const fn new() -> RPTimer<'a> {
+        RPTimer {
+            registers: TIMER_BASE,
+            client: OptionalCell::empty(),
+        }
+    }
+
+    fn enable_interrupt(&self) {
+        self.registers.inte.modify(INTE::ALARM_0::SET);
+    }
+
+    fn disable_interrupt(&self) {
+        self.registers.inte.modify(INTE::ALARM_0::CLEAR);
+    }
+
+    fn enable_timer_interrupt(&self) {
+        unsafe {
+            atomic(|| {
+                let n = cortexm0p::nvic::Nvic::new(TIMER_IRQ_0);
+                n.enable();
+            })
+        }
+    }
+
+    fn disable_timer_interrupt(&self) {
+        unsafe {
+            cortexm0p::nvic::Nvic::new(TIMER_IRQ_0).disable();
+        }
+    }
```

Comment:
flagging these as well as hopefully not needed

Reply:
We tested it and it does not work without these. I have commented these in the source.